src/ui.navigation-progress/composables/attrs.composable.js

```diff
@@ -5,72 +5,49 @@ import defaultConfig from "../configs/default.config";
 import { computed } from "vue";
 
 export function useAttrs(props) {
-  const { config, getAttrs, getColor, setColor } = useUI(defaultConfig, () => props.config);
-  const { progress, indicator, step } = config.value;
+  const { config, getAttrs, getColor, setColor, isSystemKey } = useUI(
+    defaultConfig,
+    () => props.config,
+  );
+  const attrs = {};
 
-  const cvaProgress = cva({
-    base: progress.base,
-    variants: progress.variants,
-    compoundVariants: progress.compoundVariants,
-  });
+  for (const key in defaultConfig) {
+    if (isSystemKey(key)) continue;
 
-  const cvaIndicator = cva({
-    base: indicator.base,
-    variants: indicator.variants,
-    compoundVariants: indicator.compoundVariants,
-  });
+    let classes = "";
+    let value = config.value[key];
 
-  const cvaStep = cva({
-    base: step.base,
-    variants: step.variants,
-    compoundVariants: step.compoundVariants,
-  });
+    const isVariants = value.variants || value.compoundVariants;
+    const excludeKeys = ["progress", "indicator", "step"];
 
-  const progressClasses = computed(() =>
-    setColor(
-      cvaProgress({
-        size: props.size,
-        color: getColor(props.color),
-      }),
-      props.color,
-    ),
-  );
+    if (isVariants && !excludeKeys.includes(key)) {
+      const getCVA = cva(value);
 
-  const indicatorClasses = computed(() =>
-    setColor(
-      cvaIndicator({
-        size: props.size,
-        color: getColor(props.color),
-      }),
-      props.color,
-    ),
-  );
+      classes = computed(() => getCVA({ ...props }));
+    }
 
-  const stepClasses = computed(() =>
-    setColor(
-      cvaStep({
-        size: props.size,
-        color: getColor(props.color),
-      }),
-      props.color,
-    ),
-  );
+    if (isVariants && excludeKeys.includes(key)) {
+      const getCVA = cva(value);
+
+      classes = computed(() =>
+        setColor(getCVA({ ...props, color: getColor(props.color) }), props.color),
```

Comment:
you should apply this staff to getCVA directly to allow to set color to any of config keys when user needs.